xmodule/split_test_block.py

```diff
@@ -172,10 +173,14 @@ def child_block(self):
     def child(self):
         """
         Return the user bound child block for the partition or None.
+
+        We are using try/except block here because we ran into issues with either
+        the User service being undefined or CachingDescriptorSystem has no attribute
+        when we get an icon for the split_test xblock.
         """
-        if self.child_block is not None:
+        try:
             return self.runtime.get_block_for_descriptor(self.child_block)
-        else:
+        except (NoSuchServiceError, AttributeError):
```

Comment:
```suggestion
        except (NoSuchServiceError, AttributeError):
        logging.warning("Error fetching child block: %s", e)
```
can we add some logging here?

Reply:
What this try except is for? 
`self.child_block` or `self.runtime.get_block_for_descriptor`

if for `self.child_block` I would enclose in try/except only independently around self.child_block, and keep `self.runtime.get_block_for_descriptor` outside of the try/except

Reply:
Or maybe even put try/except in the `def child_block` property if possible